assets/js/callback.js

```diff
@@ -1 +1,37 @@
-alert("authentication callback complete.")
+// alert("authentication callback complete.")
+
+function redirectUser() {
+
+  let redirectUrl
+
+  const params = new URLSearchParams(window.location.search)
+  const referringUrl = decodeURIComponent(params.get("ref"))
+  // const hasProject = decodeURIComponent(params.get("projects"))
+  const userToken = decodeURIComponent(params.get("access-token"))
+
+  if (referringUrl) {
+    redirectUrl = referringUrl
```

Comment:
good enough, though we'll want some sort of failure protection put in later so we don't wind up in a dead end or login loop.

Reply:
True, I have included a check to enure the `referringUrl` is not current page before redirecting.